cache/converter.go

```diff
@@ -91,26 +105,14 @@ func getConverter(desc ocispecs.Descriptor, compressionType compression.Type) (c
 
 type conversion struct {
 	target     compression.Type
-	decompress func(io.Reader) (cdcompression.DecompressReadCloser, error)
+	decompress func(context.Context, ocispecs.Descriptor) (io.ReadCloser, error)
```

Comment:
change it to `ReaderAt` or `ReadSeeker` instead. We should be able to define conversions for all supported compressions as a single array and then unit test them with random (tarball) data making sure that compression is lossless.

Reply:
was there an issue with this?

Reply:
> change it to ReaderAt or ReadSeeker instead.

IIUC `cdcompression.DecompressReadCloser` is not seekable. So if we want to make it seekable we need to change the current `decompress` implementation to store the decompressed blob to somewhere seekable (e.g. content store) then return the readseeker. Does this change SGTY?